### PR TITLE
[icon] fixed gradients - added uniq id for them

### PR DESCRIPTION
**Pull request by ilyabrower**
## Motivation and Context
Fixed view of icons, when we use some of them together on one page

## How has this been tested?
Code cannot be tested automatically so I have tested it only manually

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [ ] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Review by j-mnizhek — APPROVED** [review]

File: semcore/icon/svg/color/Confluence/l.svg [hunk 1]
```diff
@@ -1 +1,18 @@
...
\ No newline at end of file
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path d="M2.337 17.267c-.215.35-.457.758-.646 1.083a.663.663 0 0 0 .222.9l4.307 2.651a.663.663 0 0 0 .917-.225c.17-.288.391-.663.633-1.064 1.706-2.815 3.425-2.471 6.516-.993l4.27 2.03a.665.665 0 0 0 .892-.331l2.05-4.638a.663.663 0 0 0-.331-.868 654.128 654.128 0 0 1-4.307-2.047c-5.817-2.83-10.75-2.644-14.523 3.502Z"
+          fill="url(#confluence_l_a)"/>
+    <path d="M21.647 6.732c.216-.351.458-.762.663-1.083a.663.663 0 0 0-.225-.901l-4.307-2.65a.663.663 0 0 0-.914.225l-.636 1.063c-1.707 2.816-3.423 2.471-6.517.994L5.428 2.359a.662.662 0 0 0-.888.331L2.49 7.328a.663.663 0 0 0 .33.868c.902.424 2.697 1.269 4.307 2.047 5.814 2.806 10.747 2.62 14.52-3.511Z"
+          fill="url(#confluence_l_b)"/>
+    <defs>
+        <linearGradient id="confluence_l_a" x1="21.369" y1="23.259" x2="8.34" y2="15.769" gradientUnits="userSpaceOnUse">
```
Comment (msereniti) [inline semcore/icon/svg/color/Confluence/l.svg:7]: We need more unique ids like `intergalactic-icon-%name%-%version%`

**Review by msereniti — CHANGES_REQUESTED** [review]
Without more unique ids we can't merge it

File: semcore/icon/svg/color/Confluence/l.svg (hunk 1 above)
Comment (ilyabrower) [reply semcore/icon/svg/color/Confluence/l.svg]: added generator in icon-transform-svg

File: tools/icon-transform-svg/index.js [hunk 2]
```diff
@@ -54,6 +55,48 @@
...
+}
+
+function patchSvg($svg, fromElement, toElement, name, group) {
+  let shouldPatchId = false;
+
+  $svg.find(fromElement).each(function () {
+    const fillData = this.attribs.fill;
+
+    if (fillData) {
+      const match = fillData.match(/^url\(#(.*)+\)$/);
```
[the remark this answers is not in the document]
Comment (ilyabrower) [reply tools/icon-transform-svg/index.js]: changed regexp

**Review by msereniti — APPROVED** [review]

**Pull request merged**